vm/src/frame.rs

```diff
@@ -966,7 +970,8 @@ impl Frame {
         while let Some(block) = self.pop_block() {
             match block.typ {
                 BlockType::TryExcept { handler } => {
-                    self.push_value(exc);
+                    self.push_value(exc.clone());
```

Comment:
Its a bit weird to push the exception on the value stack as well as on the exception stack, but I guess we require this for proper value stack unwinding.